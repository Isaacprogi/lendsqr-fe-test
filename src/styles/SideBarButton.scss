@import '../styles/variables/mixins';

.show-sidebar-button{
    background-color: #9fa8b7;
    position: fixed;
    cursor: pointer;
    top: 10rem;
    color: white;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: .5rem;
    font: 900;
    font-size: 1.5rem;
    z-index: 700;
    transition: all 0.5s ease-in;
    width: .9rem;
    right: -30px;
    border-top-right-radius: 100%;
    border-bottom-right-radius: 100%;
    rotate: 180deg;

    &:hover{
        background-color: #2d4f94;
    }

    @include mediumDevices {
        display: none;
       
     }
    
      }
      
   .show-sidebar-button.visible{
            right: 0;
   }
   .show-sidebar-button.reduce{
            width: .7rem;            
   }
   .show-sidebar-button.enlarge{
            width: 2rem;          
   }