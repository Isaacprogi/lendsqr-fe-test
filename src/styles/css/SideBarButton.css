.show-sidebar-button {
  background-color: #9fa8b7;
  position: fixed;
  cursor: pointer;
  top: 10rem;
  color: white;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  padding: 0.5rem;
  font: 900;
  font-size: 1.5rem;
  z-index: 700;
  -webkit-transition: all 0.5s ease-in;
  transition: all 0.5s ease-in;
  width: 0.9rem;
  right: -30px;
  border-top-right-radius: 100%;
  border-bottom-right-radius: 100%;
  rotate: 180deg;
}
.show-sidebar-button:hover {
  background-color: #2d4f94;
}
@media screen and (min-width: 768px) {
  .show-sidebar-button {
    display: none;
  }
}

.show-sidebar-button.visible {
  right: 0;
}

.show-sidebar-button.reduce {
  width: 0.7rem;
}

.show-sidebar-button.enlarge {
  width: 2rem;
}/*# sourceMappingURL=SideBarButton.css.map */