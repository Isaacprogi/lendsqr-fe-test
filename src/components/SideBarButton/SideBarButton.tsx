import { BsChevronLeft } from 'react-icons/bs'
import { BsChevronRight } from 'react-icons/bs'
import { sideBarButtonProps } from '../../utils/types/interface_and_types'
import '../../styles/css/SideBarButton.css'

export const SideBarButton = ({showSideBar,dropDownActive,setShowSideBar}:sideBarButtonProps) => {
  return (
    <span data-testid='sidebar-button' onClick={() => setShowSideBar(!showSideBar)} className={`show-sidebar-button
    ${dropDownActive ? "" : "visible"}
    ${showSideBar? "enlarge": "reduce"}
     `}>
       {
           showSideBar ?
               <BsChevronLeft data-testid='chevron-left-icon' />
               :
               <BsChevronRight data-testid='chevron-right-icon' />
       }
   </span>
  )
}