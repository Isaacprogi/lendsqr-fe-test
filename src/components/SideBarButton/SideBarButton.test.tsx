import { fireEvent, render, screen } from '@testing-library/react';
import { SideBarButton } from './SideBarButton';
import { useState } from 'react';
import renderer from 'react-test-renderer'


describe('test for User Details Component ', () => { 
     const MySideBarComponent = () => {
        const [showSideBar,setShowSideBar] = useState<boolean>(false)
        const[dropDownActive,setDropDownActive] = useState<boolean>(false)
        return <SideBarButton showSideBar={showSideBar} dropDownActive={dropDownActive} setShowSideBar={setShowSideBar}/>
     }
    test('side-bar-button', async () => {
        render(<MySideBarComponent/>)
        const buttonElement = screen.getByTestId('sidebar-button')
        expect(buttonElement).toHaveClass('reduce')
        expect(screen.getByTestId('chevron-right-icon')).toBeInTheDocument()

        fireEvent.click(buttonElement)
        expect(buttonElement).toHaveClass('enlarge')
        expect(screen.getByTestId('chevron-left-icon')).toBeInTheDocument()
    })
    test('should match snapshot', async () => {
        const tree = renderer.create(<MySideBarComponent/>).toJSON();
        expect(tree).toMatchSnapshot();
      });
    

})
