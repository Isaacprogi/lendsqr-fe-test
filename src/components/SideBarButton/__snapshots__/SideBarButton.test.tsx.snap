// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test for User Details Component  > should match snapshot 1`] = `
<span
  className="show-sidebar-button
    visible
    reduce
     "
  data-testid="sidebar-button"
  onClick={[Function]}
>
  <svg
    data-testid="chevron-right-icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708"
      fillRule="evenodd"
    />
  </svg>
</span>
`;

exports[`test for User Details Component  should match snapshot 1`] = `
<span
  className="show-sidebar-button
    visible
    reduce
     "
  data-testid="sidebar-button"
  onClick={[Function]}
>
  <svg
    data-testid="chevron-right-icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
      fillRule="evenodd"
    />
  </svg>
</span>
`;
